> Can you provide a specific example of where you encountered these `-Wcast-qual triggers` warnings?
> 
> Please provide the complete call to `sh ./Configure` you made before this problem appeared. Also, please provide the output of `uname -a`. (We won't consider a pull request until we understand the problem fully and can reproduce it.)

The requested `uname -a` output is as follows:
```
Linux asus 6.0.10-1-default #1 SMP PREEMPT_DYNAMIC Sat Nov 26 17:59:17 UTC 2022 (582305b) x86_64 x86_64 x86_64 GNU/Linux
```

This is how I build and install Perl:
```
prefix=$HOME/software/perl-install
rm -rf "$prefix"
./Configure -d -e -s -Dusethreads -Dusedevel -Dprefix="$prefix" &&
make -j$(nproc) &&
make install &&
(cd "$prefix/bin" && ln -s perl[0-9]* perl)
make test
```

This is how I reproduce the warnings:
```
git clone https://github.com/net-snmp/net-snmp.git
cd net-snmp
PATH=$HOME/software/perl-install/bin:$PATH
export PATH
./configure --enable-developer --prefix=$HOME/net-snmp-inst --with-defaults --enable-new-features --with-perl-modules --with-default-snmp-version=2 --with-systemd --with-python-modules --with-transports="AAL5PVC IPX STD UDPshared DTLSUDP TLSTCP" --with-security-modules="usm tsm"
make -j9
```
The compiler reports the following warnings about the Perl headers while building Net-SNMP without my patch:
```
In file included from /usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/sbox32_hash.h:4,
                 from /usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_func.h:34,
                 from /usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv.h:673,
                 from /usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/perl.h:4164,
                 from snmp_perl.c:2:
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h: In function ‘zaphod32_hash_with_state’:
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:181:17: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
  181 |     U32 *state= (U32 *)state_ch;
      |                 ^
In file included from /usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_func.h:11:
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:39:35: note: in expansion of macro ‘_shifted_octet’
   39 |         #define U8TO16_LE(ptr)   (_shifted_octet(U16,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:196:33: note: in expansion of macro ‘U8TO16_LE’
  196 |             case 10: v2 += (U32)U8TO16_LE(key+8);
      |                                 ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:40:35: note: in expansion of macro ‘_shifted_octet’
   40 |                                   _shifted_octet(U16,(ptr),1, 8))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:196:33: note: in expansion of macro ‘U8TO16_LE’
  196 |             case 10: v2 += (U32)U8TO16_LE(key+8);
      |                                 ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:197:28: note: in expansion of macro ‘U8TO32_LE’
  197 |                      v1 -= U8TO32_LE(key+4);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:197:28: note: in expansion of macro ‘U8TO32_LE’
  197 |                      v1 -= U8TO32_LE(key+4);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:197:28: note: in expansion of macro ‘U8TO32_LE’
  197 |                      v1 -= U8TO32_LE(key+4);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:197:28: note: in expansion of macro ‘U8TO32_LE’
  197 |                      v1 -= U8TO32_LE(key+4);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:198:28: note: in expansion of macro ‘U8TO32_LE’
  198 |                      v0 += U8TO32_LE(key+0);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:198:28: note: in expansion of macro ‘U8TO32_LE’
  198 |                      v0 += U8TO32_LE(key+0);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:198:28: note: in expansion of macro ‘U8TO32_LE’
  198 |                      v0 += U8TO32_LE(key+0);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:198:28: note: in expansion of macro ‘U8TO32_LE’
  198 |                      v0 += U8TO32_LE(key+0);
      |                            ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:201:27: note: in expansion of macro ‘U8TO32_LE’
  201 |             case 8: v1 -= U8TO32_LE(key+4);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:201:27: note: in expansion of macro ‘U8TO32_LE’
  201 |             case 8: v1 -= U8TO32_LE(key+4);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:201:27: note: in expansion of macro ‘U8TO32_LE’
  201 |             case 8: v1 -= U8TO32_LE(key+4);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:201:27: note: in expansion of macro ‘U8TO32_LE’
  201 |             case 8: v1 -= U8TO32_LE(key+4);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:202:27: note: in expansion of macro ‘U8TO32_LE’
  202 |                     v0 += U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:202:27: note: in expansion of macro ‘U8TO32_LE’
  202 |                     v0 += U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:202:27: note: in expansion of macro ‘U8TO32_LE’
  202 |                     v0 += U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:202:27: note: in expansion of macro ‘U8TO32_LE’
  202 |                     v0 += U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:39:35: note: in expansion of macro ‘_shifted_octet’
   39 |         #define U8TO16_LE(ptr)   (_shifted_octet(U16,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:205:32: note: in expansion of macro ‘U8TO16_LE’
  205 |             case 6: v0 += (U32)U8TO16_LE(key+4);
      |                                ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:40:35: note: in expansion of macro ‘_shifted_octet’
   40 |                                   _shifted_octet(U16,(ptr),1, 8))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:205:32: note: in expansion of macro ‘U8TO16_LE’
  205 |             case 6: v0 += (U32)U8TO16_LE(key+4);
      |                                ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:206:27: note: in expansion of macro ‘U8TO32_LE’
  206 |                     v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:206:27: note: in expansion of macro ‘U8TO32_LE’
  206 |                     v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:206:27: note: in expansion of macro ‘U8TO32_LE’
  206 |                     v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:206:27: note: in expansion of macro ‘U8TO32_LE’
  206 |                     v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
libtool: compile:  gcc -I../include -I. -I../agent -I../agent/mibgroup -I/usr/include/libnl3 -I/usr/include/libnl3 -I../snmplib -I/usr/include/libnl3 -I/usr/include/libnl3 -D_GNU_SOURCE -D_ALL_SOURCE -D_THREAD_SAFE -D__EXTENSIONS__ -g -O2 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -g -O2 -Ulinux -Dlinux=linux -D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE -Wall -Wextra -Wstrict-prototypes -Wwrite-strings -Wcast-qual -Wimplicit-fallthrough -Wlogical-op -Wundef -Wno-format-truncation -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused-parameter -c mibgroup/util_funcs.c -o mibgroup/util_funcs.o >/dev/null 2>&1
libtool: compile:  gcc -I../include -I. -I../agent -I../agent/mibgroup -I/usr/include/libnl3 -I/usr/include/libnl3 -I../snmplib -I/usr/include/libnl3 -I/usr/include/libnl3 -D_GNU_SOURCE -D_ALL_SOURCE -D_THREAD_SAFE -D__EXTENSIONS__ -g -O2 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -g -O2 -Ulinux -Dlinux=linux -D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE -Wall -Wextra -Wstrict-prototypes -Wwrite-strings -Wcast-qual -Wimplicit-fallthrough -Wlogical-op -Wundef -Wno-format-truncation -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused-parameter -c agent_registry.c -o agent_registry.o >/dev/null 2>&1
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:42:35: note: in expansion of macro ‘_shifted_octet’
   42 |         #define U8TO32_LE(ptr)   (_shifted_octet(U32,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:209:27: note: in expansion of macro ‘U8TO32_LE’
  209 |             case 4: v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:43:35: note: in expansion of macro ‘_shifted_octet’
   43 |                                   _shifted_octet(U32,(ptr),1, 8)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:209:27: note: in expansion of macro ‘U8TO32_LE’
  209 |             case 4: v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
libtool: compile:  gcc -I../include -I. -I../agent -I../agent/mibgroup -I/usr/include/libnl3 -I/usr/include/libnl3 -I../snmplib -I/usr/include/libnl3 -I/usr/include/libnl3 -D_GNU_SOURCE -D_ALL_SOURCE -D_THREAD_SAFE -D__EXTENSIONS__ -g -O2 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -g -O2 -Ulinux -Dlinux=linux -D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE -Wall -Wextra -Wstrict-prototypes -Wwrite-strings -Wcast-qual -Wimplicit-fallthrough -Wlogical-op -Wundef -Wno-format-truncation -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused-parameter -c agent_trap.c -o agent_trap.o >/dev/null 2>&1
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:44:35: note: in expansion of macro ‘_shifted_octet’
   44 |                                   _shifted_octet(U32,(ptr),2,16)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:209:27: note: in expansion of macro ‘U8TO32_LE’
  209 |             case 4: v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
libtool: compile:  gcc -I../include -I. -I../agent -I../agent/mibgroup -I/usr/include/libnl3 -I/usr/include/libnl3 -I../snmplib -I/usr/include/libnl3 -I/usr/include/libnl3 -D_GNU_SOURCE -D_ALL_SOURCE -D_THREAD_SAFE -D__EXTENSIONS__ -g -O2 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -g -O2 -Ulinux -Dlinux=linux -D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE -Wall -Wextra -Wstrict-prototypes -Wwrite-strings -Wcast-qual -Wimplicit-fallthrough -Wlogical-op -Wundef -Wno-format-truncation -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused-parameter -c mibgroup/header_complex.c -o mibgroup/header_complex.o >/dev/null 2>&1
libtool: compile:  gcc -I../include -I. -I../agent -I../agent/mibgroup -I/usr/include/libnl3 -I/usr/include/libnl3 -I../snmplib -I/usr/include/libnl3 -I/usr/include/libnl3 -D_GNU_SOURCE -D_ALL_SOURCE -D_THREAD_SAFE -D__EXTENSIONS__ -g -O2 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -g -O2 -Ulinux -Dlinux=linux -D_REENTRANT -D_GNU_SOURCE -DPERL_USE_SAFE_PUTENV -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE -Wall -Wextra -Wstrict-prototypes -Wwrite-strings -Wcast-qual -Wimplicit-fallthrough -Wlogical-op -Wundef -Wno-format-truncation -Wno-missing-field-initializers -Wno-sign-compare -Wno-unused-parameter -c snmpd.c -o snmpd.o >/dev/null 2>&1
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:45:35: note: in expansion of macro ‘_shifted_octet’
   45 |                                   _shifted_octet(U32,(ptr),3,24))
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:209:27: note: in expansion of macro ‘U8TO32_LE’
  209 |             case 4: v1 -= U8TO32_LE(key+0);
      |                           ^~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:33:56: warning: cast discards ‘const’ qualifier from pointer target type [-Wcast-qual]
   33 |   #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
      |                                                        ^
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/hv_macro.h:39:35: note: in expansion of macro ‘_shifted_octet’
   39 |         #define U8TO16_LE(ptr)   (_shifted_octet(U16,(ptr),0, 0)|\
      |                                   ^~~~~~~~~~~~~~
/usr/lib/perl5/5.36.0/x86_64-linux-thread-multi/CORE/zaphod32_hash.h:212:32: note: in expansion of macro ‘U8TO16_LE’
  212 |             case 2: v0 += (U32)U8TO16_LE(key);
      |                                ^~~~~~~~~
[ ... ]
```

--- a/AUTHORS
+++ b/AUTHORS
@@ -153,6 +153,7 @@ Bah                            <bah@longitude.com>
 Barrie Slaymaker               <barries@slaysys.com>
 Barry Friedman
 Bart Kedryna                   <bkedryna@home.com>
+Bart Van Assche                <bvanassche@acm.org>
 Bas van Sisseren               <bas@quarantainenet.nl>
 Beau Cox
 Ben Carter                     <bcarter@gumdrop.flyinganvil.org>

--- a/hv_func.h
+++ b/hv_func.h
@@ -62,7 +62,7 @@
 # define PVT__PERL_HASH_SEED_BYTES (PVT__PERL_HASH_WORD_SIZE * 2)
 # define PVT__PERL_HASH_STATE_BYTES (PVT__PERL_HASH_WORD_SIZE * 4)
 # define PVT__PERL_HASH_SEED_STATE(seed,state) S_perl_siphash_seed_state(seed,state)
-# define PVT__PERL_HASH_WITH_STATE(state,str,len) S_perl_hash_siphash_1_3_with_state((state),(U8*)(str),(len))
+# define PVT__PERL_HASH_WITH_STATE(state,str,len) S_perl_hash_siphash_1_3_with_state((state),(const U8*)(str),(len))
 #elif defined(PERL_HASH_FUNC_ZAPHOD32)
 # define PERL_HASH_FUNC_DEFINE "PERL_HASH_FUNC_ZAPHOD32"
 # define PVT__PERL_HASH_FUNC "ZAPHOD32"
@@ -119,7 +119,7 @@
 
 #define PVT_PERL_HASH_WITH_STATE(state,str,len)                                            \
     (LIKELY(len <= SBOX32_MAX_LEN)                                                      \
-        ? sbox32_hash_with_state((state + PVT__PERL_HASH_STATE_BYTES),(U8*)(str),(len))    \
+        ? sbox32_hash_with_state((state + PVT__PERL_HASH_STATE_BYTES),(const U8*)(str),(len))    \
         : PVT__PERL_HASH_WITH_STATE((state),(str),(len)))
 
 #endif
@@ -127,7 +127,7 @@
 #define PERL_HASH_WITH_SEED(seed,hash,str,len) \
     (hash) = S_perl_hash_with_seed((const U8 *) seed, (const U8 *) str,len)
 #define PERL_HASH_WITH_STATE(state,hash,str,len) \
-    (hash) = PVT_PERL_HASH_WITH_STATE((state),(U8*)(str),(len))
+    (hash) = PVT_PERL_HASH_WITH_STATE((state),(const U8*)(str),(len))
 
 #define PERL_HASH_SEED_STATE(seed,state) PVT_PERL_HASH_SEED_STATE(seed,state)
 #define PERL_HASH_SEED_BYTES PVT_PERL_HASH_SEED_roundup(PVT_PERL_HASH_SEED_BYTES)

--- a/hv_macro.h
+++ b/hv_macro.h
@@ -30,7 +30,7 @@
  */
 
 #ifndef U8TO16_LE
-  #define _shifted_octet(type,ptr,idx,shift) (((type)(((U8*)(ptr))[(idx)]))<<(shift))
+  #define _shifted_octet(type,ptr,idx,shift) (((type)(((const U8*)(ptr))[(idx)]))<<(shift))
     #if defined(USE_UNALIGNED_PTR_DEREF) && (BYTEORDER == 0x1234 || BYTEORDER == 0x12345678)
         #define U8TO16_LE(ptr)   (*((const U16*)(ptr)))
         #define U8TO32_LE(ptr)   (*((const U32*)(ptr)))

--- a/inline.h
+++ b/inline.h
@@ -568,7 +568,7 @@ Perl_is_utf8_invariant_string_loc(const U8* const s, STRLEN len, const U8 ** ep)
         /* Here, we know we have at least one full word to process.  Process
          * per-word as long as we have at least a full word left */
         do {
-            if ((* (PERL_UINTMAX_T *) x) & PERL_VARIANTS_WORD_MASK)  {
+            if ((* (const PERL_UINTMAX_T *) x) & PERL_VARIANTS_WORD_MASK)  {
 
                 /* Found a variant.  Just return if caller doesn't want its
                  * exact position */
@@ -579,7 +579,7 @@ Perl_is_utf8_invariant_string_loc(const U8* const s, STRLEN len, const U8 ** ep)
 #  if   BYTEORDER == 0x1234 || BYTEORDER == 0x12345678    \
      || BYTEORDER == 0x4321 || BYTEORDER == 0x87654321
 
-                *ep = x + variant_byte_number(* (PERL_UINTMAX_T *) x);
+                *ep = x + variant_byte_number(* (const PERL_UINTMAX_T *) x);
                 assert(*ep >= s && *ep < send);
 
                 return FALSE;

--- a/sbox32_hash.h
+++ b/sbox32_hash.h
@@ -1409,7 +1409,7 @@ SBOX32_STATIC_INLINE void sbox32_seed_state128 (
     const U8 *seed_ch,
     U8 *state_ch
 ) {
-    U32 *seed= (U32 *)seed_ch;
+    const U32 *seed= (const U32 *)seed_ch;
     U32 *state= (U32 *)state_ch;
     U32 *state_cursor = state + 1;
     U32 *sbox32_end = state + 1 + (256 * SBOX32_MAX_LEN);
@@ -1458,7 +1458,7 @@ SBOX32_STATIC_INLINE U32 sbox32_hash_with_state(
     const U8 *key,
     const STRLEN key_len
 ) {
-    U32 *state= (U32 *)state_ch;
+    const U32 *state= (const U32 *)state_ch;
     U32 hash = *state;
     switch (key_len) {
         default: return zaphod32_hash_with_state(state_ch, key, key_len);

--- a/zaphod32_hash.h
+++ b/zaphod32_hash.h
@@ -178,7 +178,7 @@ U32 zaphod32_hash_with_state(
     const U8 *key,
     const STRLEN key_len
 ) {
-    U32 *state= (U32 *)state_ch;
+    const U32 *state= (const U32 *)state_ch;
     const U8 *end;
     STRLEN len = key_len;
     U32 v0= state[0];

